tests/unit/test_generate.py

```diff
@@ -0,0 +1,215 @@
+    cmd = GenerateCommand(cfg, args)
+    # git pull of https://github.com/DataDog/apigentools.git
+    cmd.pull_repository(language)
+    # make a list of the names of the directories from the repo
+    dir_contents = [el[1] for el in os.walk(temp_dir) if not el == []]
+    dirs = [directory for sublist in dir_contents for directory in sublist]
+    assert ".azure-pipelines" in dirs
+    assert "apigentools" in dirs
+    assert "tests" in dirs
+
+
+def test_write_dot_apigentools_info(tmpdir):
+    temp_dir = tmpdir.mkdir("generated")
+    repo_dir = tmpdir.mkdir("generated/DataDog")
+
+    args = flexmock(
+        action="generate",
+        additional_stamp=[],
+        api_versions=None,
+        builtin_templates=False,
+        clone_repo=False,
+        config_dir="config",
+        downstream_templates_dir="downstream-templates",
+        full_spec_file="full_spec.yaml",
+        generated_code_dir=temp_dir,
+        generated_with_image=None,
+        git_via_https=False,
+        git_via_https_installation_access_token="",
+        git_via_https_oauth_token="",
+        languages=None,
+        spec_dir="spec",
+        spec_repo_dir=".",
+        template_dir="templates",
+        verbose=False,
+        github_repo_name="repo_name",
+    )
+    with open(os.path.join(FIXTURE_DIR, "raw_dict.json"), "r") as f:
+        raw_dict = json.loads(f.read())
+    cfg = Config(raw_dict)
+    cmd = GenerateCommand(cfg, args)
+    language_cfg = LanguageConfig(
+        language="java", raw_dict=raw_dict, top_level_config=cfg
+    )
+    cmd.write_dot_apigentools_info("java")
+    with open(os.path.join(repo_dir, ".apigentools-info"), "r") as f:
+        info = f.read()
+    # testing that some of the keys are present in the .apigentools-info file
+    assert "additional_stamps" in info
+    assert "apigentools_version" in info
+    assert "codegen_version" in info
+
+
+def test_run_language_commands(tmpdir):
+    temp_dir = tmpdir.mkdir("temp_dir")
+    phase = "pre"
+    language = "java"
+    args = flexmock(
+        action="generate",
+        additional_stamp=[],
+        api_versions=None,
+        builtin_templates=False,
+        clone_repo=False,
+        config_dir="config",
+        downstream_templates_dir="downstream-templates",
+        full_spec_file="full_spec.yaml",
+        generated_code_dir=temp_dir,
+        generated_with_image=None,
+        git_via_https=False,
+        git_via_https_installation_access_token="",
+        git_via_https_oauth_token="",
+        languages=None,
+        spec_dir="spec",
+        spec_repo_dir=".",
+        template_dir="templates",
+        verbose=False,
+        github_repo_name="repo_name",
+    )
+    with open(os.path.join(FIXTURE_DIR, "raw_dict.json"), "r") as f:
+        raw_dict = json.loads(f.read())
+    cfg = Config(raw_dict)
+    cmd = GenerateCommand(cfg, args)
+    cmd.run_language_commands(language, phase, temp_dir)
+    # this runs, but I'm not sure what the commands are?
```

Comment:
Each language config, in addition to things like `repo_name` can have a `commands` section that lists some post processing commands to be done on the generated code. 

So we should add something to https://github.com/DataDog/apigentools/pull/93/files#diff-ac4598f45ec2ca4bb12011d85131f060R11 like:

```
            "commands": {
                "post": [
                    {
                        "description": "go format",
                        "commandline": [
                            "gofmt"
                        ]
                    },
                }
```
and assert that by running this function, that `gofmt` tries to be run.

Reply:
I added the config json to the fixtures directory, but I couldn't figure out an assertion that [this line in generate.py](https://github.com/DataDog/apigentools/blob/950055fa71df788f6521f2e03b1c2007e4b5a8eb/apigentools/commands/generate.py#L79) would run. As the test is currently set up, it should run "gofmt".